assets/js/react/components/common/ImportCSV/CSVDataValidator.jsx

```diff
@@ -0,0 +1,185 @@
+  const processData = (dataString) => {
+    const dataStringLines = dataString.split(/\r\n|\n/);
+    const headers = dataStringLines[0].split(
+      /,(?![^"]*"(?:(?:[^"]*"){2})*[^"]*$)/,
+    );
+
+    const list = [];
+    for (let i = 1; i < dataStringLines.length; i += 1) {
+      const row = dataStringLines[i].split(
+        /,(?![^"]*"(?:(?:[^"]*"){2})*[^"]*$)/,
+      );
+
+      if (headers && row.length === headers.length) {
+        const obj = {};
+        for (let j = 0; j < headers.length; j += 1) {
+          let cell = row[j];
+          if (cell.length > 0) {
+            if (cell[0] === '"') cell = cell.substring(1, cell.length - 1);
+            if (cell[cell.length - 1] === '"')
+              cell = cell.substring(cell.length - 2, 1);
+          }
+          if (headers[j]) {
+            obj[headers[j]] = cell;
+          }
+        }
+
+        if (Object.values(obj).filter((x) => x).length > 0) {
+          list.push(obj);
+        }
+      }
+    }
+
+    const CSVColumnsBlankLess = headers.map((column) => {
+      return column.replace(/( |\s|\r|\t|\n)+/g, '');
+    });
+
+    list.forEach((row) => {
+      const currentRow = row;
+      Object.keys(currentRow).forEach((key) => {
+        currentRow[key.replace(/( |\s|\r|\t|\n)+/g, '')] = currentRow[key];
+        if (key.includes(' ')) delete currentRow[key];
+      });
+    });
+
+    setData(list);
+    setColumns(CSVColumnsBlankLess);
+  };
+
+  const read = (file) => {
+    const reader = new window.FileReader();
+    reader.onload = (evt) => {
+      const excelData = evt.target.result;
+      const wb = XLSX.read(excelData, { type: 'binary', dateNF: 'yyyy-mm-dd' });
+
+      const sheetName = wb.SheetNames[0];
+      const ws = wb.Sheets[sheetName];
+
+      const CSVData = XLSX.utils.sheet_to_csv(ws);
+      processData(CSVData);
+    };
+    reader.readAsText(file);
+  };
+
+  const validate = () => {
+    const validatorArray = Object.keys(validator);
+    const headerErrors = {};
+    const importErrors = {};
+
+    if (
+      (selectedType === 0 &&
+        columns.length !== 28 &&
+        selectedFile !== undefined) ||
+      (selectedType === 1 &&
+        columns.length !== 25 &&
+        selectedFile !== undefined)
+    ) {
+      importErrors.delimiter = 'DELIMITER_ERROR';
+    } else {
+      columns.forEach((columnValue) => {
+        if (!validatorArray.includes(columnValue))
+          headerErrors[columnValue] = 'INVALID_HEADER';
+      });
+      if (Object.keys(headerErrors).length !== 0)
+        importErrors.header = headerErrors;
+      else {
+        for (let i = 0; i < data.length; i += 1) {
+          const currentDataRow = data[i];
+
+          if (
+            (selectedType === 0 && currentDataRow.length > 28) ||
+            (selectedType === 1 && currentDataRow.length > 25)
+          ) {
+            importErrors.delimiter = 'DELIMITER_ERROR';
+          } else {
+            for (let j = 0; j < validatorArray.length; j += 1) {
```

Comment:
A voir si c'est toujours utile en utilisant une librairie pour le csv

L'enchainement de `if` `else` `for` nesté rend le code difficile à lire. La norme est de 3/4 max
Il faut pouvoir lire le code sans savoir de quoi il s'agit et le comprendre.